bundler/lib/bundler/cli/exec.rb

```diff
@@ -18,12 +18,16 @@ def initialize(options, args)
     def run
       validate_cmd!
       SharedHelpers.set_bundle_environment
-      if bin_path = Bundler.which(cmd)
+      if (bin_path = Bundler.which(cmd))
         if !Bundler.settings[:disable_exec_load] && ruby_shebang?(bin_path)
-          return kernel_load(bin_path, *args)
+          if Gem.win_platform? # remove `.bat` suffix
+            bin_path = bin_path[0..-5]
+          end
+          kernel_load(bin_path, *args)
+        else
+          bin_path = "./" + bin_path unless File.absolute_path?(bin_path)
+          kernel_exec(bin_path, *args)
         end
-        bin_path = "./" + bin_path unless File.absolute_path?(bin_path)
-        kernel_exec(bin_path, *args)
```

Comment:
If I understand correctly, all changes above this one could be simplified to adding `file.delete_suffix!(".bat") if Gem.win_platform?` as the first line of the `kernel_load` method?